### ISSUE TITLE
Make ``superfluous-parens`` consider string combinations

### ISSUE THREAD

**Issue by DanielNoord** [PyCQA/pylint#4792]
### Current problem

Although #4784 fixes some problems with ``superfluous-parens`` it still does not consider the following cases:
```python
Z = "TestString"
X = ("Test " + "String") # Bad
Y = ("Test " + "String") in Z # Bad
assert "" + ("Version " + "String") in Z # Bad
```

This is because the current checker focuses on unnecessary parens after keywords (`if`, `else`, etc.) and only if the `(` immediately follows the keyword. See
```python
if tokens[start + 1].string != "(":
            return
``` 
on line `373` in `format.py` and the fact that we only run when Keywords are present.

Ideally, the checker would also consider string combinations and seeing if parens are necessary there.

### Desired solution

Raise ``superfluous-parens`` message on the test cases described above.

**Comment by RaenonX** [comment]
Not sure if this is exactly the same as the thing mentioned above, so I am leaving it here first instead of creating a new issue.

The following code still produces `superfluous-parens` false-positive.

```python
if not (a := dummy_func())
  return True
```

Here's the error message:
```
C0325: Unnecessary parens after 'not' keyword (superfluous-parens)
```

Versions:
```
pylint 2.9.6
astroid 2.6.6
Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
```

----------

### Syntax Reasoning

I can't use

```python
if a := not dummy_func():
  return True
```

instead of

```python
if not (a := dummy_func()):
  return True
```

because the first one only works if `dummy_func()` is returning `bool`. However, `dummy_func` is either returning `None` or some object, so I have to use the second one.

**Comment by Pierre-Sassoulas** [comment]
This should be fixed in 2.10 (that is not released yet) @RaenonX

**Comment by RaenonX** [comment]
> This should be fixed in 2.10 (that is not released yet) @RaenonX

OK, will wait for that. Thanks!

**Comment by bjd2385** [comment]
Still seeing this

**Comment by clavedeluna** [comment]
I'll work on this issue
